pkg/apis/externalHTTPEndpoint/v1alpha1/types.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"strconv"
+
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+genclient
+//+genclient:noStatus
+//+kubebuilder:object:root=true
+
+// ExternalHTTPEndpoint describes an Dapr ExternalHTTPEndpoint type to allow for external service invocation.
+type ExternalHTTPEndpoint struct {
```

Comment:
Does this need to be called "External"? If you're concerned about the name being long, how about just "HTTPEndpoint"?

Reply:
Perhaps we can go for `HTTPEndpoint` as this can be used to call HTTP services that sit even in the same pod as Dapr, and in that sense `External` doesn't necessarily convey its external to Dapr but more external to the environment, which might not always be the case.